### PR TITLE
Docs: Update Mistral AI Chat doc

### PR DESCRIPTION
**Pull request by ricken07**
- Add Chat properties options
- Fix  spring.ai.mistralai.chat.options.model descritption
- Add Mistral AI API Samples

### CODE REVIEW

File: spring-ai-docs/src/main/antora/modules/ROOT/pages/api/clients/mistralai-chat.adoc [hunk 1]
```diff
@@ -68,13 +68,18 @@ The prefix `spring.ai.mistralai.chat` is the property prefix that lets you confi
 | spring.ai.mistralai.chat.enabled | Enable MistralAI chat client.  | true
 | spring.ai.mistralai.chat.base-url   | Optional overrides the spring.ai.mistralai.base-url to provide chat specific url |  -
 | spring.ai.mistralai.chat.api-key   | Optional overrides the spring.ai.mistralai.api-key to provide chat specific api-key |  -
-| spring.ai.mistralai.chat.options.model | This is the MistralAI Chat model to use | `mistral-tiny` (the `gpt-3.5-turbo`, `gpt-4`, and `gpt-4-32k` point to the latest model versions)
+| spring.ai.mistralai.chat.options.model | This is the MistralAI Chat model to use | `mistral-tiny` (the `mistral-small`, `mistral-medium`, and `mistral-large` point to the latest model versions)
```
Comment (tzolov) [inline spring-ai-docs/src/main/antora/modules/ROOT/pages/api/clients/mistralai-chat.adoc:71]: I've added those model names as enums (and tested them): https://github.com/spring-projects/spring-ai/blob/50547188f771dc957f26823c1b167b53f782af14/models/spring-ai-mistral-ai/src/main/java/org/springframework/ai/mistralai/api/MistralAiApi.java#L699

They come from the: https://docs.mistral.ai/platform/endpoints/#mistral-ai-generative-models
So perhaps we should stick with the officially listed ones?

File: spring-ai-docs/src/main/antora/modules/ROOT/pages/api/clients/mistralai-chat.adoc [hunk 2]
```diff
@@ -68,13 +68,18 @@ The prefix `spring.ai.mistralai.chat` is the property prefix that lets you confi
...
 | spring.ai.mistralai.chat.options.safePrompt | Indicates whether to inject a security prompt before all conversations. | false
 | spring.ai.mistralai.chat.options.randomSeed | This feature is in Beta. If specified, our system will make a best effort to sample deterministically, such that repeated requests with the same seed and parameters should return the same result. | -
 | spring.ai.mistralai.chat.options.stop | Up to 4 sequences where the API will stop generating further tokens. | -
 | spring.ai.mistralai.chat.options.topP | An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens comprising the top 10% probability mass are considered. We generally recommend altering this or temperature but not both. | -
+| spring.ai.mistralai.chat.options.responseFormat | An object specifying the format that the model must output. Setting to `{ "type": "json_object" }` enables JSON mode, which guarantees the message the model generates is valid JSON.| -
```
Comment (tzolov) [inline spring-ai-docs/src/main/antora/modules/ROOT/pages/api/clients/mistralai-chat.adoc:78]: Ah, forgot about those today. Thanks for catching it.

File: spring-ai-docs/src/main/antora/modules/ROOT/pages/api/clients/mistralai-chat.adoc (hunk 1 above)
Comment (ricken07) [reply spring-ai-docs/src/main/antora/modules/ROOT/pages/api/clients/mistralai-chat.adoc:71]: Yes, they come from the official list, I'll redefine the enums instead to get closer to the value that is a valid name.

**Comment by ricken07** [comment]
Fixed the model names and used the enum in the examples

**Comment by tzolov** [comment]
Thanks @ricken07, 
But please do not merge with the upstream main. Always do pull rebase instead. e.g `git pull -r upstream main`.
Can you please copy those changes in a new clean PR.

**Pull request closed without merging**